src/Signal-Estimator-GUI/mainwindow.cpp

```diff
@@ -0,0 +1,304 @@
+        list.append("-l");
+        list.append(t);
+    }
+
+    if ((t = this->ui->Duration->cleanText()) != QString::fromStdString("10")){
+        list.append("-d");
+        list.append(t);
+    }
+
+    // both of these options have to be stdout because we use a pipe that reads from signal_estimator stdout
+    list << "--dump-output" << "-";
+    list << "--dump-input" << "-";
+
+    t = this->ui->SMA->cleanText();
+
+    list.append("--sma");
+    list.append(t);
+
+    //list.append("-f");
+
+    //list.append("text");
+
+    if ((t = this->ui->StrikePeriod->cleanText()) != QString::fromStdString("1")){
+        list.append("--strike-period");
+        list.append(t);
+    }
+
+    if ((t = this->ui->StrikeLength->cleanText()) != QString::fromStdString("0.010000")){
+        list.append("--strike-length");
+        list.append(t);
+    }
+
+    t = this->ui->StrDW->cleanText();
+    list.append("--strike-detection-window");
+    list.append(t);
+
+    t = this->ui->StrDT->cleanText();
+    list.append("--strike-detection-threshold");
+    list.append(t);
+
+    t = this->ui->SigDW->cleanText();
+    list.append("--signal-detection-window");
+    list.append(t);
+
+    t = this->ui->SigDT->cleanText();
+    list.append("--signal-detection-threshold");
+    list.append(t);
+
+    t = this->ui->GliDW->cleanText();
+    list.append("--glitch-detection-window");
+    list.append(t);
+
+    t = this->ui->GliDT->cleanText();
+    list.append("--glitch-detection-threshold");
+    list.append(t);
+
+    if ((t = this->ui->MVFD->cleanText()) != QString::fromStdString("64")){
+        list.append("--dump-frame");
+        list.append(t);
+    }
+
+    if ((t = this->ui->DR->cleanText()) != QString::fromStdString("10")){
+        list.append("--dump-rounding");
+        list.append(t);
+    }
+
+    return list;
+}
+
+void MainWindow::run_estimator(){
+        QStringList args = this->set_up_program();
+
+        this->set_update_plots(true); // must be true to update graphs
+
+        // clear old data when the start button is pressed
+        in_data.clearBuf();
+        out_data.clearBuf();
+
+        // reset graphs
+        this->ui->OutputSig->updateAxes();
+        this->ui->OutputSig->replot();
+
+        this->proc = startSignalEstimator(args);
+        // proc emits error signal
+        this->proc->connect(this->proc,SIGNAL(error(QProcess::ProcessError)),this,SLOT(checkProc()));
+
+        QTimer* time = new QTimer(0);
+
+        this->timer = time; // timer for ui graphing
+
+        timer->setInterval(20);
+        timer->connect(timer, SIGNAL(timeout()),this,SLOT(update_graphs()));
+        this->proc->connect(this->proc,SIGNAL(readyReadStandardOutput()), this, SLOT(read_graph_data()));
+        //this->proc->connect(this->proc,SIGNAL(finished(int)), this, SLOT(on_StopButton_clicked()));
+
+        if (this->proc->open(QProcess::ReadOnly) == false) // failing? to open signal-est
```

Comment:
Can we move open() call to startSignalEstimator? It may return nullptr when failed to open, and we'll check it here.